test/__init__.py

```diff
@@ -19,3 +22,23 @@ def __exit__(self, *args):
     def captured(self):
         self.buffer.seek(0, io.SEEK_SET)
         return self.buffer.read()
+
+
+def reset_tweak_changes(f):
```

Comment:
I'm not sure a function decorator is superior to a with-statement context.  The with-statement context can be applied to a narrow scope of code, whereas the function decorator has to be applied to an entire function.

However, I will defer to your judgement. :)